app/scripts/controllers/transactions/index.js

```diff
@@ -2110,6 +2127,18 @@ export default class TransactionController extends EventEmitter {
       transactionContractMethod = transactions[id]?.contractMethodName;
     }
 
+    const replacedTxMeta = this._getTransactionByHash(replacedBy);
+
+    let transactionReplaced;
+    if (extraParams?.dropped) {
+      transactionReplaced = TRANSACTION_TYPES.OTHER;
```

Comment:
I don't like the addition of this 'OTHER' transaction_type constant. What I would prefer is creating a new constant:

```js
const TRANSACTION_REPLACEMENT_METHODS = {
  RETRY: TRANSACTION_TYPES.RETRY,
  CANCEL: TRANSACTION_TYPES.CANCEL,
  SAME_NONCE: 'other',
};
```

Using a different constant reduces the chances of confusing future contributors who might think that 'other' is a viable transaction type when it is only there for metrics.

Reply:
I agree with you definitely, I will create a new constant `TRANSACTION_REPLACEMENT_METHODS`! Thank you for this good suggestion @brad-decker!

Reply:
@brad-decker Done [here](https://github.com/MetaMask/metamask-extension/pull/15383/commits/1c9525d3f99a0d7e073e759aa96624b20fc8c209).